src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String DELIMITER = ",";
+    public static final String OPERATION = "buy";
+    private int supplyAmount = 0;
+    private int buyAmount = 0;
+    private final StringBuilder stringBuilder = new StringBuilder();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromFile = new File(fromFileName);
+        final File toFile = new File(toFileName);
+
+        readDataFromTheFile(fromFile);
+
+        int result = supplyAmount - buyAmount;
+
+        stringBuilder.setLength(0);
+        stringBuilder.append("supply,").append(supplyAmount).append(System.lineSeparator())
+                .append("buy,").append(buyAmount).append(System.lineSeparator())
+                .append("result,").append(result);
+
+        writeToTheFile(toFile);
+    }
+
+    public void readDataFromTheFile(File fromFile) {
+        List<String> fileData;
+        try {
+            fileData = Files.readAllLines(fromFile.toPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Cant read from file", e);
+        }
+
+        for (String data : fileData) {
+            String[] values = data.split(DELIMITER);
+
+            String operation = values[0];
+            String amount = values[1];
+
+            if (operation.equals(OPERATION)) {
+                buyAmount += Integer.parseInt(amount);
+            } else {
+                supplyAmount += Integer.parseInt(amount);
+            }
+        }
+    }
 
+    public void writeToTheFile(File toFile) {
```

Comment:
The method 'writeToTheFile' should be private if it is only used within this class.